### PR TITLE
#45_iowa_crash

### LINKED ISSUE #45: As a user, I can self-destruct on the station iowa map without the server segfaulting

### PR DESCRIPTION
**Pull request by solarisstar**
Deleted call to useent in self_destrucT_think that caused recursive use calls. Doesn't seem to have hurt the functionality but stopped iowa from crashing

fixes https://github.com/solarisstar/rpgxEF/issues/45

This solution felt really hacky, but it seemed to be the only way to prevent recursive use calls. It's a pretty nasty case since the recursion occurs four calls deep into the stack. (Four different functions keep looping forever until the server gives up, ejects all the clients, and segfaults) 

I tested self destruct on ctf_voy1 and on staiton_iowa and the functionality all seems to work as expected without any crashing. I'm not sure if this has broken anything, but everything appears to be fine.

### CODE REVIEW

**Comment by MJLHThomassen** [comment]
Its possible for the self-destruct entity (as wel as others) to fire another entity when itself is triggered, thats what the ent->target does. It is probably not breaking anything because the functionality is not used on the map. So for some reason, the target field is set, else it wouldn't recurse.

It's probably best to find out why this is recursing and prevent that instead of doing what you changed. 

If we decide to go with this solution, we need to modify the QUAKED comment of the entity and also adjust it in the rpg-x_entities.def

**Comment by solarisstar** [comment]
It might be something on the .map file which we cannot access. For instance, what if the mapper set tow entities to trigger each other?

**Comment by solarisstar** [comment]
All right, looks like I found the real culprit. You'll laugh.

File: code/game/g_target.c [hunk 1]
```diff
@@ -2812,7 +2812,7 @@ void target_selfdestruct_think(gentity_t *ent) {
         }
         //let's hear it
         G_AddEvent(ent, EV_GLOBAL_SOUND, G_SoundIndex("sound/weapons/explosions/explode2.wav"));
-        //let's be shakey for a sec... I hope lol ^^
+        //let's be shakey for a sec... I hope lol ^_^
```
Comment (solarisstar) [inline code/game/g_target.c:2815]: Couldn't resist.

Comment (MJLHThomassen) [inline code/game/g_target.c:2815]: haha

**Comment by MJLHThomassen** [comment]
Lol, so target would sometimes be '\0' then? How is it then that other entities don't have this issue?

**Comment by solarisstar** [comment]
It's possible that they're usually NULL, but the mapper may have set it to the empty string rather than just NULL.

In either case, picking entities that have an empty string as a target should be undefined behavior and not something we're allowing going forward.

**Comment by MJLHThomassen** [comment]
Agreed, either NULL or "\0" shouldn't be accepted as targetname. 

Strange how this would cause the loop though, because G_Find shoulnn't be finding any target if the field was left empty right?

**Comment by solarisstar** [comment]
Not at a computer right now but G_Find uses this ridiculous offset logic to select the field to compare against the search value. I don't think there's any logic in there to prevent comparisons of the empty string. They only seem to protect against NULL pointers.

**Comment by solarisstar** [comment]
We good for this PR?

**Comment by MJLHThomassen** [comment]
Yes, sorry about that. This seems good.

**Pull request merged**